src/chessboard/components/Arrows.tsx

```diff
@@ -0,0 +1,90 @@
+import { Fragment } from "react";
+
+import { getRelativeCoords } from "../functions";
+import { useChessboard } from "../context/chessboard-context";
+import { Square } from "../types";
+
+export const Arrows = () => {
+  const {
+    arrows,
+    newArrow,
+    boardOrientation,
+    boardWidth,
+
+    customArrowColor,
+  } = useChessboard();
+
+  const arrowsList = [...arrows, newArrow].filter(Boolean) as Square[][];
+  return (
+    <svg
+      width={boardWidth}
+      height={boardWidth}
+      style={{
+        position: "absolute",
+        top: "0",
+        left: "0",
+        pointerEvents: "none",
+        zIndex: "10",
+      }}
+    >
+      {arrowsList.map((arrow, i) => {
+        const from = getRelativeCoords(boardOrientation, boardWidth, arrow[0]);
+        const to = getRelativeCoords(boardOrientation, boardWidth, arrow[1]);
+        let ARROW_LENGTH_REDUCER = boardWidth / 32;
+
+        const isArrowActive = i === arrows.length;
+        // if there are differnet arrows targeting same square make thier length a bit shorter
+        if (
+          arrows.some(
+            (restArrow) =>
+              restArrow[0] !== arrow[0] && restArrow[1] === arrow[1]
+          ) &&
+          !isArrowActive
+        ) {
+          ARROW_LENGTH_REDUCER = boardWidth / 16;
+        }
```

Comment:
Can I see what it looks like without making their lengths shorter when targeting the same square, this functionality seemed a bit off, so I'm curious to see what it's like as normal and then if I agree it needs shortening we can add this back

Reply:
https://github.com/Clariity/react-chessboard/assets/42891870/c6f4d7d6-5627-432e-a30e-b724b65563f8

Reply:
I think you've made a great decision here then with the shortening on the same square, I'll take some time soon to give the code a good review but I think this should be merged in soon, great job

Reply:
Thanks